File: solutions/Golang/intersection_of_two_arrays_two_pointer.go
package golang
import "sort"

func intersection(nums1 []int, nums2 []int) []int {
	sort.Ints(nums1)
	sort.Ints(nums2)
	result := []int{}
















}	return result	}		}			j++		} else {			i++		} else if nums1[i] < nums2[j] {			j++			i++			}				result = append(result, nums1[i])			if len(result) == 0 || result[len(result)-1] != nums1[i] {		if nums1[i] == nums2[j] {	for i < len(nums1) && j < len(nums2) {	i, j := 0, 0

File: solutions/Java/intersection_of_two_arrays_two_pointer.java
import java.util.*;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        List<Integer> result = new ArrayList<>();
        int i = 0, j = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] == nums2[j]) {
                if (result.isEmpty() || result.get(result.size() - 1) != nums1[i])
                    result.add(nums1[i]);
                i++; j++;
            } else if (nums1[i] < nums2[j]) {
                i++;
            } else {
                j++;
            }
        }
        return result.stream().mapToInt(x -> x).toArray();
    }
}
